pkg/infra/httpclient/httpclientprovider/http_client_provider.go

```diff
@@ -2,17 +2,21 @@ package httpclientprovider
 
 import (
 	"fmt"
+	"net"
+	"net/http"
+	"time"
 
 	sdkhttpclient "github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
 	"github.com/grafana/grafana/pkg/infra/httpclient"
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/mwitkow/go-conntrack"
 )
 
 var newProviderFunc = sdkhttpclient.NewProvider
 
 // New creates a new HTTP client provider with pre-configured middlewares.
-func New(cfg *setting.Cfg) httpclient.Provider {
+func New(cfg *setting.Cfg, connName string) httpclient.Provider {
```

Comment:
This is created once when Grafana starts. Think you want separate name per data source? See other comment